### PR TITLE
add completion handler option to moveOverlay

### PR DESCRIPTION
**Pull request by joustava**
Added the completion handler as described in #19 

TODO:
- [x] Use this commit in own project.
- [x] Check if current unit test is sufficient.
- [x] Remove parameter from callback?

### CODE REVIEW

**Comment by joustava** [comment]
@gaetanzanella your feedback is welcome. I'll update the PR when needed.

File: Source/Classes/Internal/HeightContraintOverlayTranslationController.swift [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 import UIKit
 
-class HeightContrainstOverlayTranslationController: OverlayTranslationController {
+class HeightContraintsOverlayTranslationController: OverlayTranslationController {
```
Comment (gaetanzanella) [inline Source/Classes/Internal/HeightContraintOverlayTranslationController.swift]: The file's name is wrong. Both should be named `HeightConstraintOverlayTranslationController`.

File: Source/Classes/Internal/HeightContraintOverlayTranslationController.swift [hunk 2]
```diff
@@ -92,7 +92,7 @@ class HeightContrainstOverlayTranslationController: OverlayTranslationController
         moveOverlay(toNotchAt: index, velocity: velocity, animated: true)
     }
 
-    func moveOverlay(toNotchAt index: Int, velocity: CGPoint, animated: Bool) {
+    func moveOverlay(toNotchAt index: Int, velocity: CGPoint, animated: Bool, completion: @escaping ((UIViewAnimatingPosition) -> Void) = {_ in }) {
```
Comment (gaetanzanella) [inline Source/Classes/Internal/HeightContraintOverlayTranslationController.swift]: Please remove the position parameter

File: Source/Classes/Internal/HeightContraintOverlayTranslationController.swift (hunk 1 above)
Comment (gaetanzanella) [inline Source/Classes/Internal/HeightContraintOverlayTranslationController.swift]: The file's name is wrong. Both should be named HeightConstraintOverlayTranslationController.

File: Source/Classes/Internal/OverlayTranslationController.swift [hunk 3]
```diff
@@ -26,7 +26,7 @@ protocol OverlayTranslationController: class {
     func isDraggable(at point: CGPoint, in coordinateSpace: UICoordinateSpace) -> Bool
     func overlayHasReachedANotch() -> Bool
 
-    func moveOverlay(toNotchAt index: Int, velocity: CGPoint, animated: Bool)
+    func moveOverlay(toNotchAt index: Int, velocity: CGPoint, animated: Bool, completion: @escaping (UIViewAnimatingPosition) -> Void)
```
Comment (gaetanzanella) [inline Source/Classes/Internal/OverlayTranslationController.swift]: Same comment

File: Source/Classes/Internal/HeightContraintOverlayTranslationController.swift [hunk 4]
```diff
@@ -121,7 +121,9 @@ class HeightContrainstOverlayTranslationController: OverlayTranslationController
             willReachNotchAt: translationEndNotchIndex,
             transitionCoordinator: coordinator
         )
+        animator.addCompletion?(completion)
         animator.startAnimation()
+
```
Comment (gaetanzanella) [inline Source/Classes/Internal/HeightContraintOverlayTranslationController.swift]: Please remove line break

File: Source/Classes/Internal/HeightContraintOverlayTranslationController.swift [hunk 5]
```diff
@@ -121,7 +121,9 @@ class HeightContrainstOverlayTranslationController: OverlayTranslationController
             willReachNotchAt: translationEndNotchIndex,
             transitionCoordinator: coordinator
         )
+        animator.addCompletion?(completion)
```
Comment (gaetanzanella) [inline Source/Classes/Internal/HeightContraintOverlayTranslationController.swift]: The completion block should be also called is the move is not animated

File: Source/Classes/SpringOverlayTranslationAnimationController.swift [hunk 6]
```diff
@@ -42,6 +42,7 @@ public class SpringOverlayTranslationAnimationController: OverlayAnimatedTransit
             response: response,
             mass: mass
         )
+
```
Comment (gaetanzanella) [inline Source/Classes/SpringOverlayTranslationAnimationController.swift]: Please remove line break

File: Tests/OverlayNotchHeightTest.swift [hunk 7]
```diff
@@ -103,11 +103,16 @@ class OverlayNotchHeightTest: QuickSpec {
...
                 overlayContainer.moveOverlay(toNotchAt: notch.rawValue, animated: true) { position in
-                    // Not sure about this
+                    called = true
+                    expectation.fulfill()
+
                 }
-                //overlayContainer.view.layoutIfNeeded()
-                //expect(overlay.view.frame.height).to(equal(delegate.height(for: notch)))
+                self.waitForExpectations(timeout: 5, handler: nil)
+                expect(called).to(beTrue())
```
Comment (gaetanzanella) [inline Tests/OverlayNotchHeightTest.swift]: What about `waitUntilDone`?

Here is an example from Nimble [readme](https://github.com/Quick/Nimble#asynchronous-expectations)

```
waitUntil { done in
    ocean.goFish { success in
        expect(success).to(beTrue())
        done()
    }
}
```

File: Source/Classes/OverlayContainerViewController.swift [hunk 8]
```diff
@@ -136,11 +136,11 @@ public class OverlayContainerViewController: UIViewController {
     /// - parameter index: The index of the target notch.
     /// - parameter animated: Defines either the transition should be animated or not.
     ///
-    public func moveOverlay(toNotchAt index: Int, animated: Bool) {
+    public func moveOverlay(toNotchAt index: Int, animated: Bool, completion: @escaping ((UIViewAnimatingPosition) -> Void) = {_ in }) {
```
Comment (gaetanzanella) [inline Source/Classes/OverlayContainerViewController.swift]: Please make it optional and nil by default

**Comment by gaetanzanella** [comment]
Thanks!

**Comment by joustava** [comment]
You're welcome! Thanks again for making this library.

**Comment by gaetanzanella** [comment]
Yes I will!

**Pull request merged**